bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -262,6 +258,32 @@ public void partClosed(IWorkbenchPart part) {
 		public void partOpened(IWorkbenchPart part) {
 			// Do nothing
 		}
+
+		private void adaptToPartActivationChange() {
+			if (getShell() == null || targetPart.getSite().getShell() == null) {
+				return;
+			}
+
+			getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+
+			targetPart.getSite().getShell().setActive();
```

Comment:
`targetPart.getSite().getPart().setFocus();` - why not `targetPart.setFocus();` ?

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/ad63e43557ee5c01ce8857d21d4dc5436905f9bb